app/src/main/java/org/yuttadhammo/BodhiTimer/TimerActivity.kt

```diff
@@ -321,6 +322,17 @@ class TimerActivity : AppCompatActivity(), View.OnClickListener, OnSharedPrefere
         return super.onKeyDown(keycode, e)
     }
 
+    override fun onBackPressed(){
+        // NOTE: Let back key act as home key, so auto-restarting
+        // feature can work well when user press back key.
+        Toast.makeText(this, getString(R.string.run_app_in_background),
+                       Toast.LENGTH_SHORT).show()
+        val intent = Intent(Intent.ACTION_MAIN)
+        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        intent.addCategory(Intent.CATEGORY_HOME)
+        startActivity(intent)
+    }
+
```

Comment:
I don't understand the connection between the back key and the auto-repeat timer.. Can you explain?

Reply:
Ok, I get what you mean. But back pressing is only one possibilty by which the apps activity can get killed by. 
So the point is not to avoid the backpress, but to make sure the timerservice works independently of the Activities lifecycle.

Reply:
I'm glad to hear that you've recovered, good news!

```
Ok, I get what you mean. But back pressing is only one possibilty by which the apps activity can get killed by.
So the point is not to avoid the backpress, but to make sure the timerservice works independently of the Activities lifecycle.
```

I do not know much about android development, may be we have better way to solve this problem :-)

Reply:
Yes, I am feeling much better now :)
I have merged the other PR, and added some changes of my own. 
Can you check if AutoRestart works now? It does for me. 
If not, send the logs!

Reply:
i will test tomorrow, thanks very much